### PR TITLE
bwapi.ini additions and environment variable overrides

### PR DESCRIPTION
**Pull request by blole**
These are some changes that I've made and liked, they are mostly stand alone but I figured you'd prefer a single pull request.

The biggest one is being able to override bwapi.ini settings via environment variables, which could be very handy for a tournament with multiple starcraft instances running on the same computer; but also nice when launching starcraft with debugging from visual studio. This feature should probably be documented somewhere, I've described how it works in 7007f66.

### CODE REVIEW

File: bwapi/BWAPI/Source/BWAPI/BulletImpl.h [hunk 1]
```diff
@@ -40,7 +40,7 @@ namespace BWAPI
       BW::CBullet* getRawData() const;
       void        saveExists();
 
-      BulletData  data = {};
+      BulletData  data;
```
Comment (heinermann) [inline bwapi/BWAPI/Source/BWAPI/BulletImpl.h]: Why are we no longer initializing this data structure?

**Comment by blole** [comment]
Oh, right, there's a difference. Well, doing the `= {}` won't compile for me. But initializing it in the BulletImpl constructor does work. Maybe we can do that instead?
I changed the same thing in PlayerData and RegionData as well in 0f8c61d.

**Comment by heinermann** [comment]
Why won't it compile? Which compiler are you using?

**Comment by blole** [comment]
Visual Studio 2013 (v120)
```
C:\tm\bwapi\bwapi\BWAPI\Source\BWAPI\BulletImpl.h(43): fatal error C1001: An internal error has occurred in the compiler.
  (compiler file 'msc1.cpp', line 1325)
   To work around this problem, try simplifying or changing the program near the locations listed above.
  Please choose the Technical Support command on the Visual C++
   Help menu, or open the Technical Support help file for more information (..\Shared\BulletShared.cpp)
  INTERNAL COMPILER ERROR in 'C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\CL.exe'
      Please choose the Technical Support command on the Visual C++
      Help menu, or open the Technical Support help file for more information
```

I replaced `= {}` with `= BulletData()` in #631. Now it compiles for me and from what I understand, that should also zero initialize it, according to http://en.cppreference.com/w/cpp/language/value_initialization

**Comment by heinermann** [comment]
OK. Add this workaround to your latest PR and I'll merge it in.

IIRC not zero-initializing this structure will cause problems, especially on game start or in the BWAPI Client.

**Comment by heinermann** [comment]
What's weird is that other developers can compile it just fine with VC++ 2013 Express, and the build server also compiles it.

**Pull request merged**